commons-datastore/commons-datastore-solr/src/main/java/org/opencb/commons/datastore/solr/SolrManager.java

```diff
@@ -353,6 +329,42 @@ public void close() throws IOException {
         }
     }
 
+    public SolrClient newSolrClient(int timeout) {
+        SolrClient solrClient;
+        if (hosts.get(0).startsWith("http")) {
+            if (hosts.size() == 1) {
+                // Single HTTP endpoint.
+                solrClient = new HttpSolrClient.Builder(hosts.get(0)).build();
+                ((HttpSolrClient) solrClient).setRequestWriter(new BinaryRequestWriter());
+                ((HttpSolrClient) solrClient).setSoTimeout(timeout);
+            } else {
+                // Use a LoadBalancer if there are multiple http hosts
+                solrClient = new LBHttpSolrClient.Builder().withBaseSolrUrls(hosts.toArray(new String[0])).build();
+
+                ((LBHttpSolrClient) solrClient).setRequestWriter(new BinaryRequestWriter());
+                ((LBHttpSolrClient) solrClient).setSoTimeout(timeout);
+            }
+        } else {
+            // If the provided hosts are not http, assume zookeeper hosts like HOST:PORT
+            // This client will use Zookeeper to discover Solr endpoints for SolrCloud collections, and then use the
+            // LBHttpSolrClient to issue requests.
+            if (isCloud()) {
+                solrClient = new CloudSolrClient.Builder().withZkHost(hosts).build();
+
+                ((CloudSolrClient) solrClient).setRequestWriter(new BinaryRequestWriter());
+                ((CloudSolrClient) solrClient).setSoTimeout(timeout);
+            } else {
+                throw new IllegalArgumentException("Can not initialize SolrManager from Zookeeper host not in Cloud mode");
+            }
+        }
+
+        // Sanity check
+        if (solrClient == null) {
```

Comment:
You can make "solrClient" final, so this check would not be needed
The compiler itself will complain if the variable might be unassigned at this point.